tests/test__mongomock.py

```diff
@@ -2403,6 +2403,25 @@ def test__aggregate_project_id_0(self):
         pipeline = [{'$project': {'_id': 0}}]
         self.cmp.compare_ignore_order.aggregate(pipeline)
 
+    def test__aggregate_project_array_subfield(self):
+        self.cmp.do.insert_many([
+            {'a': [{'b': 1, 'c': 2, 'd': 3}], 'e': 4},
```

Comment:
Please add one test with an array with a mix of dicts and int  `{'a': [{'b': 1}, 1, {'c': 2}, {'d': 3}, 'c']}`

Reply:
indeed, i even forgot to test the multiple matching elements in an array case.